runtime/onert/core/src/util/shapeinf/ReduceSum.cc

```diff
@@ -0,0 +1,155 @@
+/* Copyright (c) 2017 The TensorFlow Authors. All Rights Reserved.
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "util/ShapeInference.h"
+
+namespace onert
+{
+namespace shape_inference
+{
+
+ir::Shape inferReduceShapes(const ir::Shape &input_shape, const std::vector<int> &axes,
+                            bool keep_dims)
+{
+  int num_axis = axes.size();
+  int input_num_dims = input_shape.rank();
+  if (input_num_dims == 0)
+  {
+    ir::Shape out_shape(0);
+    return out_shape;
+  }
+  if (keep_dims)
+  {
+    ir::Shape out_shape;
+    for (int idx = 0; idx < input_num_dims; ++idx)
+    {
+      bool is_axis = false;
+      for (int axis_idx = 0; axis_idx < num_axis; ++axis_idx)
+      {
+        if (axes[axis_idx] == idx || axes[axis_idx] + input_num_dims == idx)
+        {
+          is_axis = true;
+          break;
+        }
+      }
+      if (is_axis)
+      {
+        out_shape.append(1);
+      }
+      else
+      {
+        out_shape.append(input_shape.dim(idx));
+      }
+    }
+    return out_shape;
+  }
+  else
+  {
+    // Calculates size of reducing axis.
+    int num_reduce_axis = num_axis;
+    for (int i = 0; i < num_axis; ++i)
+    {
+      int current = axes[i];
+      if (current < 0)
+      {
+        current += input_num_dims;
+      }
+      assert(0 <= current && current < rank);
```

Comment:
when I pull this PR, I meet the error. 'rank' was not declared in this scope.
what do you mean this "rank"?

Reply:
Thank you. I modified it.